src/handlers/account-deletion-processor-handler.ts

```diff
@@ -73,7 +74,8 @@ async function updateDeleteStatusId(userId: string) {
     await ddbService.updateDeleteStatus(userId);
   } catch (error) {
     logger.error(`${LOGS_PREFIX_SENSITIVE_INFO} Error updating account ${userId}`, { error });
-    logAndPublishMetric(MetricNames.MARK_AS_DELETED_FAILED);
+    addMetric(MetricNames.MARK_AS_DELETED_FAILED);
+    metric.publishStoredMetrics();
```

Comment:
wondering if we could do try {} catch (e) {} finally { metric.publishStoredMetrics(); } so as to not do metric.publishStoredMetrics(); in try and catch separately

Reply:
just a suggestion ;-)

Reply:
we could.. but then this would mean that if we are processing multiple records successfully, we would publish once per record, as opposed to one unique call when Promise.all resolved... whereas this way we publish once when an error is thrown, which interrupts the flow (so the call after promise.All doesn't execute bc the error is thrown)

Reply:
hope this makes sense @TaherKapasi :)